diff --git a/lib/rules/consistent-return.js b/lib/rules/consistent-return.js
index eca8ec3d5..58a225203 100644
--- a/lib/rules/consistent-return.js
+++ b/lib/rules/consistent-return.js
@@ -85,11 +85,11 @@ module.exports = {
 			 * When unreachable, all paths are returned or thrown.
 			 */
 			if (
-				!funcInfo.hasReturnValue ||
-				!isAnySegmentReachable(funcInfo.currentSegments) ||
-				astUtils.isES5Constructor(node) ||
-				isClassConstructor(node)
-			) {
+			!funcInfo.hasReturnValue ||
+			!isAnySegmentReachable(funcInfo.currentSegments) ||
+			astUtils.isES5Constructor(node) ||
+			isClassConstructor(node)
+		) {
 				return;
 			}
 
