diff --git a/lib/rules/array-callback-return.js b/lib/rules/array-callback-return.js
index 019a60e84..8f8708580 100644
--- a/lib/rules/array-callback-return.js
+++ b/lib/rules/array-callback-return.js
@@ -10,7 +10,7 @@
 //------------------------------------------------------------------------------
 
 const astUtils = require("./utils/ast-utils");
-const { isAnySegmentReachable } = require("./utils/code-path-utils");
+const { isAnySegmentReachable } = require("../shared/segment-analysis");
 
 //------------------------------------------------------------------------------
 // Helpers
diff --git a/lib/rules/consistent-return.js b/lib/rules/consistent-return.js
index 58a225203..6407739b7 100644
--- a/lib/rules/consistent-return.js
+++ b/lib/rules/consistent-return.js
@@ -10,12 +10,26 @@
 
 const astUtils = require("./utils/ast-utils");
 const { upperCaseFirst } = require("../shared/string-utils");
-const { isAnySegmentReachable } = require("./utils/code-path-utils");
 
 //------------------------------------------------------------------------------
 // Helpers
 //------------------------------------------------------------------------------
 
+/**
+ * Checks all segments in a set and returns true if all are unreachable.
+ * @param {Set<CodePathSegment>} segments The segments to check.
+ * @returns {boolean} True if all segments are unreachable; false otherwise.
+ */
+function areAllSegmentsUnreachable(segments) {
+	for (const segment of segments) {
+		if (segment.reachable) {
+			return false;
+		}
+	}
+
+	return true;
+}
+
 /**
  * Checks whether a given node is a `constructor` method in an ES6 class
  * @param {ASTNode} node A node to check
@@ -85,11 +99,11 @@ module.exports = {
 			 * When unreachable, all paths are returned or thrown.
 			 */
 			if (
-			!funcInfo.hasReturnValue ||
-			!isAnySegmentReachable(funcInfo.currentSegments) ||
-			astUtils.isES5Constructor(node) ||
-			isClassConstructor(node)
-		) {
+				!funcInfo.hasReturnValue ||
+				areAllSegmentsUnreachable(funcInfo.currentSegments) ||
+				astUtils.isES5Constructor(node) ||
+				isClassConstructor(node)
+			) {
 				return;
 			}
 
diff --git a/lib/rules/getter-return.js b/lib/rules/getter-return.js
index 28c10fa25..687863c81 100644
--- a/lib/rules/getter-return.js
+++ b/lib/rules/getter-return.js
@@ -10,7 +10,7 @@
 //------------------------------------------------------------------------------
 
 const astUtils = require("./utils/ast-utils");
-const { isAnySegmentReachable } = require("./utils/code-path-utils");
+const { isAnySegmentReachable } = require("../shared/segment-analysis");
 
 //------------------------------------------------------------------------------
 // Helpers
diff --git a/lib/rules/no-fallthrough.js b/lib/rules/no-fallthrough.js
index 5a2caae0e..aede08d8b 100644
--- a/lib/rules/no-fallthrough.js
+++ b/lib/rules/no-fallthrough.js
@@ -9,7 +9,7 @@
 //------------------------------------------------------------------------------
 
 const { directivesPattern } = require("../shared/directives");
-const { isAnySegmentReachable } = require("./utils/code-path-utils");
+const { isAnySegmentReachable } = require("../shared/segment-analysis");
 
 //------------------------------------------------------------------------------
 // Helpers
diff --git a/lib/rules/no-unreachable-loop.js b/lib/rules/no-unreachable-loop.js
index 200da8bd6..5a228c54e 100644
--- a/lib/rules/no-unreachable-loop.js
+++ b/lib/rules/no-unreachable-loop.js
@@ -5,7 +5,7 @@
 
 "use strict";
 
-const { isAnySegmentReachable } = require("./utils/code-path-utils");
+const { isAnySegmentReachable } = require("../shared/segment-analysis");
 
 //------------------------------------------------------------------------------
 // Helpers
diff --git a/lib/rules/no-unreachable.js b/lib/rules/no-unreachable.js
index 093d9f744..7be94bb3f 100644
--- a/lib/rules/no-unreachable.js
+++ b/lib/rules/no-unreachable.js
@@ -4,8 +4,6 @@
  */
 "use strict";
 
-const { isAnySegmentReachable } = require("./utils/code-path-utils");
-
 //------------------------------------------------------------------------------
 // Helpers
 //------------------------------------------------------------------------------
@@ -25,6 +23,21 @@ function isInitialized(node) {
 	return Boolean(node.init);
 }
 
+/**
+ * Checks all segments in a set and returns true if all are unreachable.
+ * @param {Set<CodePathSegment>} segments The segments to check.
+ * @returns {boolean} True if all segments are unreachable; false otherwise.
+ */
+function areAllSegmentsUnreachable(segments) {
+	for (const segment of segments) {
+		if (segment.reachable) {
+			return false;
+		}
+	}
+
+	return true;
+}
+
 /**
  * The class to distinguish consecutive unreachable statements.
  */
@@ -141,7 +154,7 @@ module.exports = {
 			if (
 				node &&
 				(node.type === "PropertyDefinition" ||
-				!isAnySegmentReachable(currentCodePathSegments))
+					areAllSegmentsUnreachable(currentCodePathSegments))
 			) {
 				// Store this statement to distinguish consecutive statements.
 				if (range.isEmpty) {
diff --git a/lib/rules/no-useless-return.js b/lib/rules/no-useless-return.js
index a2f0380d5..21a73443d 100644
--- a/lib/rules/no-useless-return.js
+++ b/lib/rules/no-useless-return.js
@@ -10,7 +10,7 @@
 
 const astUtils = require("./utils/ast-utils"),
 	FixTracker = require("./utils/fix-tracker");
-const { isAnySegmentReachable } = require("./utils/code-path-utils");
+const { isAnySegmentReachable } = require("../shared/segment-analysis");
 
 //------------------------------------------------------------------------------
 // Helpers
diff --git a/lib/rules/utils/code-path-utils.js b/lib/rules/utils/code-path-utils.js
deleted file mode 100644
index 5646466f7..000000000
--- a/lib/rules/utils/code-path-utils.js
+++ /dev/null
@@ -1,22 +0,0 @@
-/**
- * @fileoverview Code path related utilities.
- */
-
-"use strict";
-
-/**
- * Checks all segments in a set and returns true if any are reachable.
- * @param {Set<CodePathSegment>} segments The segments to check.
- * @returns {boolean} `true` if any segment is reachable; `false` otherwise.
- */
-function isAnySegmentReachable(segments) {
-	for (const segment of segments) {
-		if (segment.reachable) {
-			return true;
-		}
-	}
-
-	return false;
-}
-
-module.exports = { isAnySegmentReachable };
